src/picongpu/include/simulationControl/MySimulation.hpp

```diff
@@ -335,6 +341,17 @@ class MySimulation : public SimulationHelper<simDim>
 
         /* add CUDA streams to the StreamController for concurrent execution */
         Environment<>::get().StreamController().addStreams(6);
+
+        // create factory for the random number generator
+        this->rngFactory = new RNGFactory(Environment<simDim>::get().SubGrid().getLocalDomain().size);
+        // init factory
+        PMacc::GridController<simDim>& gridCon = PMacc::Environment<simDim>::get().GridController();
+        this->rngFactory->init(gridCon.getScalarPosition());
+
+#if(ENABLE_SYNCHROTRON_PHOTONS == 1)
```

Comment:
Could you please use the trait `hasFlag()` and scan all species if the feature is enabled and depending on this result initialize the `synchrotronFunctions()`.

This avoid code flow control with macros.